src/main/lib/files.js

```diff
@@ -72,13 +72,21 @@ function showSaveDialog(focusedWindow) {
  * @param {BrowserWindow} win
  */
 export function loadFile(filePath, win, isNew = false) {
-  const payload = {
+  let payload = {
     type: ArchiveTypes.FILE,
     path: normalizePath(filePath),
     isNew
   };
   if (path.extname(filePath).toLowerCase() !== '.bcup') {
-    return;
+    if (!isNew) {
+      return;
+    }
+
+    // automatically add the .bcup extension to new archive.
+    payload = {
+      ...payload,
+      path: normalizePath(filePath + '.bcup')
```

Comment:
This rewrite of `payload` isn't needed, it would be fine to just set `path`, leaving the definition as `const`.